ResetPasswordBundle/src/Persistence/PasswordResetRequestRepositoryInterface.php

```diff
@@ -1,16 +1,32 @@
 <?php
 
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
 namespace SymfonyCasts\Bundle\ResetPassword\Persistence;
 
+use Symfony\Component\Security\Core\User\UserInterface;
 use SymfonyCasts\Bundle\ResetPassword\Model\PasswordResetRequestInterface;
 
+/**
+ * @author Jesse Rushlow <jr@rushlow.dev>
+ * @author Ryan Weaver <weaverryan@gmail.com>
+ */
 interface PasswordResetRequestRepositoryInterface
 {
-    public function createPasswordResetRequest(object $user, \DateTimeInterface $expiresAt, string $selector, string $hashedToken): PasswordResetRequest;
+    public function createPasswordResetRequest(UserInterface $user, \DateTimeImmutable $expiresAt, string $selector, string $hashedToken): PasswordResetRequestInterface;
 
     public function persistPasswordResetRequest(PasswordResetRequestInterface $passwordResetRequest);
 
     public function findPasswordResetRequest(string $selector): ?PasswordResetRequestInterface;
 
-    public function getMostRecentNonExpiredRequestDate(object $user): ?\DateTimeInterface;
+    public function getMostRecentNonExpiredRequestDate(UserInterface $user): ?\DateTimeImmutable;
+
+    public function removeResetRequest(PasswordResetRequestInterface $resetRequest): void;
```

Comment:
name the arg`$passwordResetRequest` and the method `removePasswordResetRequest` (purely for consistency with `persistPasswordResetRequest`)

Reply:
We *may* also need an additional method - like `removeExpiredResetRequests()` for the "purging old records" command/garbage control. But we could add that in a different PR.

Reply:
done

will introduce some form of garbage collection/expired tokens separetly